xls/dslx/ir_convert/extract_conversion_order.cc

```diff
@@ -661,7 +661,18 @@ static void RemoveFunctionDuplicates(std::vector<ConversionRecord>* ready) {
       bool either_is_parametric =
           function_cr.f()->IsParametric() || subject_cr.f()->IsParametric();
 
-      if (same_fns && !either_is_proc_instance_fn && !either_is_parametric) {
+      bool both_is_parametric =
+          function_cr.f()->IsParametric() && subject_cr.f()->IsParametric();
+
+      bool equal_parametric_env = false;
+      if (both_is_parametric) {
+        equal_parametric_env =
+            function_cr.parametric_env() == subject_cr.parametric_env();
+      }
+
+      if (same_fns && !either_is_proc_instance_fn &&
+          (!either_is_parametric ||
+           (both_is_parametric && equal_parametric_env))) {
```

Comment:
you should need to test for `both_is_parametric` here again (since `equal_parametric_env` would be false if it is false)

Reply:
True